### PR TITLE
fix: Add additional log context when reference can not be found

### PR DESCRIPTION
**Pull request by shawner18**
I ran into an issue where the BOM was failing to generate due to dependency resolution issues. The current logging was not clear enough to identify the source of the issue, since we had many projects and many dependencies.

**Original output**
Dependency ([2.29.0, )) with version range ([2.29.0, )) did not resolve to a specific version.
Unable to locate valid bom ref for my.library [2.29.0, )

**Enhanced output**
Dependency (my.library) with version range ([2.29.0, )) referenced by (Name:my.other.library Version:1.18.26) in did not resolve to a specific version.
Unable to locate valid bom ref for my.library [2.29.0, ) referenced in (Name: my.other.library Version: 1.18.26)

### CODE REVIEW

**Comment by adcorduneanu** [comment]
Any fix for this one?

**Comment by CodeTigerCloud** [comment]
Hello shawner18,

I'd like to merge this, but the sign-off is missing. 
Could you add it please?

See here: https://github.com/CycloneDX/cyclonedx-dotnet/pull/662/checks?check_run_id=11826932593

**Comment by shawner18** [comment]
> Hello shawner18,
> 
> I'd like to merge this, but the sign-off is missing. Could you add it please?
> 
> See here: https://github.com/CycloneDX/cyclonedx-dotnet/pull/662/checks?check_run_id=11826932593

Done, sorry about that

**Review by CodeTigerCloud — CHANGES_REQUESTED** [review]
Yeah, sorry :(
I just saw this now

File: CycloneDX/Services/ProjectAssetsFileService.cs [hunk 1]
```diff
@@ -122,7 +122,7 @@ private static void ResolveDependencyVersionRanges(HashSet<NugetPackage> runtime
                         else
                         {
                             // This should not happen, since all dependencies are resolved to a specific version.
-                            Console.Error.WriteLine($"Dependency ({dependency.Key}) with version range ({dependency.Value}) did not resolve to a specific version.");
+                            Console.Error.WriteLine($"Dependency ({dependency.Key}) with version range ({dependency.Value}) referenced by (Name:{runtimePackage.Name} Version:{runtimePackage.Version}) in did not resolve to a specific version.");
```
[the remark this answers is not in the document]
Comment (shawner18) [reply CycloneDX/Services/ProjectAssetsFileService.cs]: fixed the text

**Review by CodeTigerCloud — APPROVED** [review]

**Comment by Kubana** [comment]
Hi I am trying to generate BOM file for my .sln and I keep getting as well similar error:
"Unable to locate valid bom ref for NETStandard.Library [2.0.1, )" 
And couple of these: "Dependency (NETStandard.Library) with version range ([2.0.1, )) referenced by (Name:Microsoft.Azure.Storage.Queue Version:11.1.7) did not resolve to a specific version."

The thing is when I go .csproj after .csproj everything goes smooth I mean the latter error keeps popping up but the BOM gets generated regardless. And another thing is when I am looking for something that references NETStandard.Library with version 2.0.1 - there is nothing ... I don't know maybe I'm just blind or looking somewhere else

**Pull request merged**